packages/mco/components/mco-dashboard/data-policy/status-card/status-card.tsx

```diff
@@ -57,6 +59,30 @@ const getDRCombinedStatus = (
   return getOperatorHealthState(worstPhase, !csvHubLoaded, csvHubError);
 };
 
+const getClustersOperatorHealth = (
+  csvManagedData: PrometheusResponse,
+  drClusterAppsMap: DrClusterAppsMap
+) => {
+  const clusterWiseHealth: ClusterWiseHealth =
+    csvManagedData?.data?.result?.reduce((acc, item) => {
+      const cluster = item?.metric.cluster;
+      let count = acc?.[cluster] || 0;
+      acc =
+        drClusterAppsMap.hasOwnProperty(cluster) &&
+        !!clusterOperators.find((operator: string) =>
+          item?.metric.name.startsWith(operator)
+        )
+          ? { ...acc, [cluster]: item?.value[1] !== '1' ? count : count + 1 }
+          : acc;
+      return acc;
+    }, {});
+  return Object.keys(clusterWiseHealth)?.every(
+    (cluster) => clusterWiseHealth?.[cluster] === clusterOperators.length
+  )
+    ? HealthState.OK
+    : HealthState.ERROR;
```

Comment:
why so much extra calculation ?? is there any issue with this ??
```suggestion
    const isAnyUnhealthy: boolean = !!csvManagedData?.data?.result?.find(
      (item) =>
        drClusterAppsMap.hasOwnProperty(item?.metric.cluster) &&
        !!clusterOperators.find((operator: string) =>
          item?.metric.name.startsWith(operator)
        ) &&
        item?.value[1] !== '1'
    );
    return isAnyUnhealthy ? HealthState.ERROR : HealthState.OK;
```

Reply:
it will stop at very first unhealthy operator it will find (in any cluster)...
for your part it will always calculate to the end and then determine the final outcome...

Reply:
if u think in future we will need names of all unhealthy clusters, we can make that change in future (it is a small change only)... why do extra calculations now, when we don;t need it...

Reply:
this logic has a huge problem since csvManagedData is a metric, Anycase if the operator is not reporting the metrics (Some problem in installation) then it will consider only the reported metrics and mark the status as healthy.

In the cluster-wise dashboard we are checking metrics for each operator by a specific name, if no metrics exist for the particular operator then there it will display as unhealthy. 

Basically, over all status card is displaying healthy, clusterwise is displaying unhealthy, it's a mismatch in logic. What i did, per cluster wise, i am looking at metrics for all the cluster operators.

Reply:
make sense !!